applications/GeoMechanicsApplication/custom_utilities/process_parameters.h

```diff
@@ -0,0 +1,37 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+#pragma once
+
+#include "includes/kratos_parameters.h"
+
+#include <string>
+
+namespace Kratos
+{
+
+struct ProcessParameters
```

Comment:
@avdg81 For this one I got this SQ issue, which I don't completely understand, because I didn't implement a move or assignment operator here:
https://sonarcloud.io/project/issues?resolved=false&sinceLeakPeriod=true&types=CODE_SMELL&branch=geo%2Fadd-process-factory-to-settlement&id=KratosGeomechanics&open=AYqz1Y3GQOYbbtPOkeNJ

Reply:
As discussed, this is because Parameter is not 'no exept', which is out of scope to fix in this PR

Reply:
I believe that this issue is caused by the fact that the move operator of class `Parameters` is not `noexcept`. (Which is rather unfortunate and not in line with the C++ Core Guidelines.) Since you have a data member of type `Parameters`, the compiler cannot generate a `noexcept` move operator for your new class. If we want to resolve this, we need to correct it in the Kratos core.

Reply:
Agreed, this issue is outside the scope of this PR.